### ISSUE TITLE
404 error in console

### ISSUE THREAD

**Issue by Naunis10** [Mario-Duarte/ImageZoom#11]
When using this component in my app, I noticed an strange error in the console: a call to {pageURL}/null. Looking at the demo page I see a similar error appears there:

![image](https://github.com/Mario-Duarte/react-image-zooom/assets/131315556/acb9ec25-67b7-405e-8de6-e3e7667ba8b6)

Please fix.

**Comment by stepan-twnty** [comment]
@Mario-Duarte any news?

**Comment by Mario-Duarte** [comment]
I can recreate the issue but have not had a chance to look into it.

**Comment by stepan-twnty** [comment]
@Mario-Duarte I take a look at the code and the problem with 
<img width="537" alt="image" src="https://github.com/Mario-Duarte/react-image-zooom/assets/124367863/c6703004-5581-4e7c-96b7-11c0a487de40">

On the first load, src is null and the browser requests this "null". It looks like adding a small guard should resolve the problem

```
 <Figure
        id={id}
        className={[figureClass, figureZoomed, className].join(" ")}
        style={{
          backgroundImage: `url( ${zoomed === '0' ? imgData : ''} )`,
          backgroundSize: zoom + "%",
          backgroundPosition: position,
        }}
        onClick={handleClick}
        onMouseMove={handleMove}
        onMouseLeave={handleLeave}
        onTouchStart={handleClick}
        onTouchMove={handleMove}
        onTouchEnd={handleLeave}
      >
       // Here is guard for preventing null request
        {imgData && <Img 
          id="imageZoom"
          src={imgData}
          alt={alt}
          style={{ opacity: zoomed }}
          width={width}
          height={height}
        />}
      </Figure>
    ```

**Comment by stepan-twnty** [comment]
hey @Mario-Duarte 

I would appreciate it if you check it because due to these issues, I have a lot of problems with my flow.

Thx in advance

**Comment by Mario-Duarte** [comment]
> @Mario-Duarte I take a look at the code and the problem with <img alt="image" width="537" src="https://private-user-images.githubusercontent.com/124367863/329812946-c6703004-5581-4e7c-96b7-11c0a487de40.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTU2MDM0MDgsIm5iZiI6MTcxNTYwMzEwOCwicGF0aCI6Ii8xMjQzNjc4NjMvMzI5ODEyOTQ2LWM2NzAzMDA0LTU1ODEtNGU3Yy05NmI3LTExYzBhNDg3ZGU0MC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTEzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUxM1QxMjI1MDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lM2UzZmNmZWY1Njg1NGI1MDEwNzc2ZWE1Yjg0ZTM5YWEzMWNhZjU5ZDA2OTdiZGZiZjg1MWMzYzFhYTg1MzNhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.i2f2DIFI9Vhbr9I0B1vNnOJB82RcxAbHSCebPyQcI8c">
> 
> On the first load, src is null and the browser requests this "null". It looks like adding a small guard should resolve the problem
> 
> ```
>  <Figure
>         id={id}
>         className={[figureClass, figureZoomed, className].join(" ")}
>         style={{
>           backgroundImage: `url( ${zoomed === '0' ? imgData : ''} )`,
>           backgroundSize: zoom + "%",
>           backgroundPosition: position,
>         }}
>         onClick={handleClick}
>         onMouseMove={handleMove}
>         onMouseLeave={handleLeave}
>         onTouchStart={handleClick}
>         onTouchMove={handleMove}
>         onTouchEnd={handleLeave}
>       >
>        // Here is guard for preventing null request
>         {imgData && <Img 
>           id="imageZoom"
>           src={imgData}
>           alt={alt}
>           style={{ opacity: zoomed }}
>           width={width}
>           height={height}
>         />}
>       </Figure>
>     ```     
>  
> ```

This is a recent change that was added to enable dynamically setting the source of the image and this issue is from before this.

**Comment by Mario-Duarte** [comment]
> hey @Mario-Duarte
> 
> I would appreciate it if you check it because due to these issues, I have a lot of problems with my flow.
> 
> Thx in advance

I understand the frustration, I will have a look when I can. In the meantime feel free to have a look at the code and open a pull request if you would like to make changes.

**Comment by stepan-twnty** [comment]
@Mario-Duarte done https://github.com/Mario-Duarte/react-image-zooom/pull/16

**Comment by Mario-Duarte** [comment]
> @Mario-Duarte done #16

That's great, I will review it later today when I got a minute

**Comment by Mario-Duarte** [comment]
This has now been fixed on [v1.3.5](https://github.com/Mario-Duarte/react-image-zooom/releases/tag/1.3.5)
`npm update react-image-zooom` to update to latest version.
